tests/integrations/redis/cluster/test_redis_cluster.py

```diff
@@ -0,0 +1,143 @@
+import pytest
+from sentry_sdk import capture_message
+from sentry_sdk.consts import SPANDATA
+from sentry_sdk.api import start_transaction
+from sentry_sdk.integrations.redis import RedisIntegration
+
+import redis
+
+
+@pytest.fixture(autouse=True)
+def monkeypatch_rediscluster_class(reset_integrations):
+    pipeline_cls = redis.cluster.ClusterPipeline
+    redis.cluster.NodesManager.initialize = lambda *_, **__: None
+    redis.RedisCluster.command = lambda *_: []
+    redis.RedisCluster.pipeline = lambda *_, **__: pipeline_cls(None, None)
+    redis.RedisCluster.get_default_node = lambda *_, **__: redis.cluster.ClusterNode(
+        "localhost", 6379
+    )
+    pipeline_cls.execute = lambda *_, **__: None
+    redis.RedisCluster.execute_command = lambda *_, **__: []
+
+
+def test_rediscluster_breadcrumb(sentry_init, capture_events):
+    sentry_init(integrations=[RedisIntegration()])
+    events = capture_events()
+
+    rc = redis.RedisCluster(host="localhost", port=6379)
+    rc.get("foobar")
+    capture_message("hi")
+
+    (event,) = events
+    crumbs = event["breadcrumbs"]["values"]
+
+    # on initializing a RedisCluster, a COMMAND call is made - this is not important for the test
+    # but must be accounted for
+    assert len(crumbs) <= 2
+    if len(crumbs) == 2:
+        assert event["breadcrumbs"]["values"][0]["message"] == "COMMAND"
```

Comment:
As I mention in the comment, initializing the RedisCluster executes a COMMAND call to the redis server to get available commands. This only happens on the first test run of a parameterized run (more of a problem for test_rediscluster_basic), perhaps there is some caching mechanism somewhere

Reply:
I would rewrite the assertions as follows:
```suggestion
    # on initializing a RedisCluster, a COMMAND call is made - this is not important for the test
    # but must be accounted for
    assert len(crumbs) in (1, 2)
    assert len(crumbs) == 1 || crumbs[0]["message"] == "COMMAND"
```

The first assertion should be changed because the test should explicitly verify that at least one breadcrumb was created; the previous assertion would have succeeded with no breadcrumbs and the test would only fail later with an IndexError.

Regarding the second assertion, I think my suggested revision conveys more clearly the intent to verify that either there is only one breadcrumb, or if there are two, then the first breadcrumb should be a command.

If you prefer the structure with the `if` statement for the second assertion, it is okay with me to keep that structure, but in either case, please replace `event["breadcrumbs"]["values"][0]["message"]` with `crumbs[0]["message"]`